OneSignalSDK/onesignal/notifications/src/main/java/com/onesignal/notifications/internal/listeners/NotificationListener.kt

```diff
@@ -98,10 +103,16 @@ internal class NotificationListener(
             NotificationHelper.getCampaignNameFromNotification(openResult.notification),
         )
 
+        // Handle the first element in the data array as the latest notification
```

Comment:
Simplification suggestion, we could move this outcome logic to a private function, early returns to simply this logic as well.